plugins/hydra_submitit_launcher/hydra_plugins/hydra_submitit_launcher/submitit_launcher.py

```diff
@@ -111,7 +111,7 @@ def launch(
             **{
                 f"{self._EXECUTOR}_{x}": y
                 for x, y in params.items()
-                if x in specific_init_keys
+                if x in specific_init_keys and y is not None
```

Comment:
I would rather avoid it.
The config should contain what we need and nothing more.

Reply:
There's a second issue here where the AutoExecutor won't accept `None` values because of some of type checking it does. If we simply don't pass them it works as expected which is why I'm filtering them out here

Reply:
> The config should contain what we need and nothing more.

I agree with that.

>  AutoExecutor won't accept None values

we can also fix this on AutoExecutor side, see eg: https://github.com/facebookincubator/submitit/pull/1607

Reply:
> The config should contain what we need and nothing more.

The config it still correct though right? It's just that hydra marks optional parameters with `None` if they aren't used and submitit wants unused optionals to not be passed and this code reconciles the difference

> we can also fix this on AutoExecutor side

Do you want to though? I think the contract with users right now is very clear that optional parameters are not passed rather than assigned `None`

Reply:
> The config it still correct though right? It's just that hydra marks optional parameters with None if they aren't used and submitit wants unused optionals to not be passed and this code reconciles the difference

If the config contains None when it shouldn't it's not a correct config.
We either make None a legitimate input  or remove it from the config (probably by moving the field to the appropriate subclass).

> Do you want to though? I think the contract with users right now is very clear that optional parameters are not passed rather than assigned None

My opinion is that that we should be using the SlurmExecutor directly but I am yielding to the submitit team here.
What to do or not to do in AutoExecutor is their call.

Reply:
> If the config contains None when it shouldn't it's not a correct config.
We either make None a legitimate input or remove it from the config (probably by moving the field to the appropriate subclass).

Right so either we make an exception for this plugin because `None` means something different for submitit, or we fix on the autoexecutor side, I'm happy to push on both fronts

> My opinion is that that we should be using the SlurmExecutor directly but I am yielding to the submitit team here.
What to do or not to do in AutoExecutor is their call.

Agreed, using the executor directly makes the most sense to me but deferring to them seems like the right call here.